### PR TITLE
Replace shell-based repo test script with a Python version

### PR DESCRIPTION
**Pull request by aerique**
Product owners, could you please take a looksy and perhaps test this for your specific product?

The focus of this PR is replacing the functionality of the shell-based script.  The automatic building and running of the Docker images is a nice extra but needs more work (error reporting for example).

I'll rebase and turn this PR into one commit.

This PR removes the shell-based repo script and replaces it with a Python version that should be easier to understand, read and modify.

The shell-based version was getting a little long in the tooth :-D

- compared all generated files for all releases of the Python version against the old ones
    - some intended changes and whitespace differences were spotted
    - in the meantime there have been some changes to the script, sorry.  i will do this again but it is very time intensive
- tested a few releases ("steekproef") and these worked
    - `auth-42` CentOS 7
    - `rec-45` Debian Buster
    - `dnsdist-15` Ubuntu Focal
- tested all releases using the build-in build & run functionality
    - using ` for RELEASE in auth-41 auth-42 auth-43 auth-44 auth-master rec-42 rec-43 rec-44 rec-45 rec-master dnsdist-15 dnsdist-16 dnsdist-master; do ./generate-repo-files.py --test $RELEASE; done` (see comment below for results)

### CODE REVIEW

**Comment by aerique** [comment]
```
$ for RELEASE in auth-41 auth-42 auth-43 auth-44 auth-master rec-42 rec-43 rec-44 rec-45 rec-master dnsdist-15 dnsdist-16 dnsdist-master; do ./generate-repo-files.py --test $RELEASE; done
Building Docker image using Dockerfile.auth-41.centos-7...
Error building auth-41.centos-7: 1
Skipping Dockerfile.auth-41.centos-7 due to build error: 1
Building Docker image using Dockerfile.auth-41.ubuntu-bionic...
Error building auth-41.ubuntu-bionic: 100
Skipping Dockerfile.auth-41.ubuntu-bionic due to build error: 100
Test done.
- failed builds:
    - ('Dockerfile.auth-41.centos-7', 1)
    - ('Dockerfile.auth-41.ubuntu-bionic', 100)
Building Docker image using Dockerfile.auth-42.centos-7...
Running Docker container tagged auth-42.centos-7...
Building Docker image using Dockerfile.auth-42.centos-8...
Running Docker container tagged auth-42.centos-8...
Building Docker image using Dockerfile.auth-42.debian-buster...
Running Docker container tagged auth-42.debian-buster...
Building Docker image using Dockerfile.auth-42.raspbian-buster...
Running Docker container tagged auth-42.raspbian-buster...
Building Docker image using Dockerfile.auth-42.ubuntu-bionic...
Running Docker container tagged auth-42.ubuntu-bionic...
Test done.
Building Docker image using Dockerfile.auth-43.centos-6...
Error building auth-43.centos-6: 139
Skipping Dockerfile.auth-43.centos-6 due to build error: 139
Building Docker image using Dockerfile.auth-43.centos-7...
Running Docker container tagged auth-43.centos-7...
Building Docker image using Dockerfile.auth-43.centos-8...
Running Docker container tagged auth-43.centos-8...
Building Docker image using Dockerfile.auth-43.debian-buster...
Running Docker container tagged auth-43.debian-buster...
Building Docker image using Dockerfile.auth-43.raspbian-buster...
Running Docker container tagged auth-43.raspbian-buster...
Building Docker image using Dockerfile.auth-43.ubuntu-bionic...
Running Docker container tagged auth-43.ubuntu-bionic...
Building Docker image using Dockerfile.auth-43.ubuntu-focal...
Running Docker container tagged auth-43.ubuntu-focal...
Test done.
- failed builds:
    - ('Dockerfile.auth-43.centos-6', 139)
Building Docker image using Dockerfile.auth-44.centos-7...
Running Docker container tagged auth-44.centos-7...
Building Docker image using Dockerfile.auth-44.centos-8...
Running Docker container tagged auth-44.centos-8...
Building Docker image using Dockerfile.auth-44.debian-buster...
Running Docker container tagged auth-44.debian-buster...
Building Docker image using Dockerfile.auth-44.raspbian-buster...
Running Docker container tagged auth-44.raspbian-buster...
Building Docker image using Dockerfile.auth-44.ubuntu-bionic...
Running Docker container tagged auth-44.ubuntu-bionic...
Building Docker image using Dockerfile.auth-44.ubuntu-focal...
Running Docker container tagged auth-44.ubuntu-focal...
Test done.
Building Docker image using Dockerfile.auth-master.centos-6...
Error building auth-master.centos-6: 139
Skipping Dockerfile.auth-master.centos-6 due to build error: 139
Building Docker image using Dockerfile.auth-master.centos-7...
Running Docker container tagged auth-master.centos-7...
Building Docker image using Dockerfile.auth-master.centos-8...
Running Docker container tagged auth-master.centos-8...
Building Docker image using Dockerfile.auth-master.debian-buster...
Running Docker container tagged auth-master.debian-buster...
Building Docker image using Dockerfile.auth-master.raspbian-buster...
Running Docker container tagged auth-master.raspbian-buster...
Building Docker image using Dockerfile.auth-master.ubuntu-bionic...
Running Docker container tagged auth-master.ubuntu-bionic...
Building Docker image using Dockerfile.auth-master.ubuntu-focal...
Running Docker container tagged auth-master.ubuntu-focal...
Test done.
- failed builds:
    - ('Dockerfile.auth-master.centos-6', 139)
Building Docker image using Dockerfile.rec-42.centos-7...
Running Docker container tagged rec-42.centos-7...
Building Docker image using Dockerfile.rec-42.centos-8...
Running Docker container tagged rec-42.centos-8...
Building Docker image using Dockerfile.rec-42.debian-buster...
Running Docker container tagged rec-42.debian-buster...
Building Docker image using Dockerfile.rec-42.raspbian-buster...
Running Docker container tagged rec-42.raspbian-buster...
Building Docker image using Dockerfile.rec-42.ubuntu-bionic...
Running Docker container tagged rec-42.ubuntu-bionic...
Test done.
Building Docker image using Dockerfile.rec-43.centos-7...
Running Docker container tagged rec-43.centos-7...
Building Docker image using Dockerfile.rec-43.centos-8...
Running Docker container tagged rec-43.centos-8...
Building Docker image using Dockerfile.rec-43.debian-buster...
Running Docker container tagged rec-43.debian-buster...
Building Docker image using Dockerfile.rec-43.raspbian-buster...
Running Docker container tagged rec-43.raspbian-buster...
Building Docker image using Dockerfile.rec-43.ubuntu-bionic...
Running Docker container tagged rec-43.ubuntu-bionic...
Building Docker image using Dockerfile.rec-43.ubuntu-focal...
Running Docker container tagged rec-43.ubuntu-focal...
Test done.
Building Docker image using Dockerfile.rec-44.centos-7...
Running Docker container tagged rec-44.centos-7...
Building Docker image using Dockerfile.rec-44.centos-8...
Running Docker container tagged rec-44.centos-8...
Building Docker image using Dockerfile.rec-44.debian-buster...
Running Docker container tagged rec-44.debian-buster...
Building Docker image using Dockerfile.rec-44.raspbian-buster...
Running Docker container tagged rec-44.raspbian-buster...
Building Docker image using Dockerfile.rec-44.ubuntu-bionic...
Running Docker container tagged rec-44.ubuntu-bionic...
Building Docker image using Dockerfile.rec-44.ubuntu-focal...
Running Docker container tagged rec-44.ubuntu-focal...
Test done.
Building Docker image using Dockerfile.rec-45.centos-7...
Running Docker container tagged rec-45.centos-7...
Building Docker image using Dockerfile.rec-45.centos-8...
Running Docker container tagged rec-45.centos-8...
Building Docker image using Dockerfile.rec-45.debian-buster...
Running Docker container tagged rec-45.debian-buster...
Building Docker image using Dockerfile.rec-45.raspbian-buster...
Error building rec-45.raspbian-buster: 100
Skipping Dockerfile.rec-45.raspbian-buster due to build error: 100
Building Docker image using Dockerfile.rec-45.ubuntu-bionic...
Running Docker container tagged rec-45.ubuntu-bionic...
Building Docker image using Dockerfile.rec-45.ubuntu-focal...
Running Docker container tagged rec-45.ubuntu-focal...
Test done.
- failed builds:
    - ('Dockerfile.rec-45.raspbian-buster', 100)
Building Docker image using Dockerfile.rec-master.centos-7...
Running Docker container tagged rec-master.centos-7...
Building Docker image using Dockerfile.rec-master.centos-8...
Running Docker container tagged rec-master.centos-8...
Building Docker image using Dockerfile.rec-master.debian-buster...
Running Docker container tagged rec-master.debian-buster...
Building Docker image using Dockerfile.rec-master.raspbian-buster...
Running Docker container tagged rec-master.raspbian-buster...
Building Docker image using Dockerfile.rec-master.ubuntu-bionic...
Running Docker container tagged rec-master.ubuntu-bionic...
Building Docker image using Dockerfile.rec-master.ubuntu-focal...
Running Docker container tagged rec-master.ubuntu-focal...
Test done.
Building Docker image using Dockerfile.dnsdist-15.centos-7...
Running Docker container tagged dnsdist-15.centos-7...
Building Docker image using Dockerfile.dnsdist-15.centos-8...
Running Docker container tagged dnsdist-15.centos-8...
Building Docker image using Dockerfile.dnsdist-15.debian-buster...
Running Docker container tagged dnsdist-15.debian-buster...
Building Docker image using Dockerfile.dnsdist-15.raspbian-buster...
Running Docker container tagged dnsdist-15.raspbian-buster...
Building Docker image using Dockerfile.dnsdist-15.ubuntu-bionic...
Running Docker container tagged dnsdist-15.ubuntu-bionic...
Building Docker image using Dockerfile.dnsdist-15.ubuntu-focal...
Running Docker container tagged dnsdist-15.ubuntu-focal...
Test done.
Building Docker image using Dockerfile.dnsdist-16.centos-7...
Running Docker container tagged dnsdist-16.centos-7...
Building Docker image using Dockerfile.dnsdist-16.centos-8...
Running Docker container tagged dnsdist-16.centos-8...
Building Docker image using Dockerfile.dnsdist-16.debian-buster...
Running Docker container tagged dnsdist-16.debian-buster...
Building Docker image using Dockerfile.dnsdist-16.raspbian-buster...
Error building dnsdist-16.raspbian-buster: 100
Skipping Dockerfile.dnsdist-16.raspbian-buster due to build error: 100
Building Docker image using Dockerfile.dnsdist-16.ubuntu-bionic...
Running Docker container tagged dnsdist-16.ubuntu-bionic...
Building Docker image using Dockerfile.dnsdist-16.ubuntu-focal...
Running Docker container tagged dnsdist-16.ubuntu-focal...
Test done.
- failed builds:
    - ('Dockerfile.dnsdist-16.raspbian-buster', 100)
Building Docker image using Dockerfile.dnsdist-master.centos-7...
Running Docker container tagged dnsdist-master.centos-7...
Building Docker image using Dockerfile.dnsdist-master.centos-8...
Running Docker container tagged dnsdist-master.centos-8...
Building Docker image using Dockerfile.dnsdist-master.debian-buster...
Running Docker container tagged dnsdist-master.debian-buster...
Building Docker image using Dockerfile.dnsdist-master.raspbian-buster...
Running Docker container tagged dnsdist-master.raspbian-buster...
Building Docker image using Dockerfile.dnsdist-master.ubuntu-bionic...
Running Docker container tagged dnsdist-master.ubuntu-bionic...
Building Docker image using Dockerfile.dnsdist-master.ubuntu-focal...
Running Docker container tagged dnsdist-master.ubuntu-focal...
Test done.
```

File: build-scripts/docker/repo-test/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+In the directory of this README and `generate-repo-files.py` file do:
+
+- `python3 -m venv venv`
+- `bash`
+- `source venv/bin/activate`
+- `pip install --upgrade pip`
+- `pip install -r requirements.txt`
+- `./generate-repo-files.py auth-41`
+  - where `rec-45` is an example release, do `.generate-repo-files.py`
```
Comment (rgacogne) [inline build-scripts/docker/repo-test/README.md:32]:
```suggestion
  - where `rec-45` is an example release, do `.generate-repo-files.py`
```
```suggestion
  - where `auth-41` is an example release, do `.generate-repo-files.py`
```

**Review by rgacogne — APPROVED** [review]
I have not tested the script but the general logic looks good :)

File: build-scripts/docker/repo-test/README.md (hunk 1 above)
Comment (omoerbeek) [inline build-scripts/docker/repo-test/README.md:32]:
```suggestion
  - where `rec-45` is an example release, do `./generate-repo-files.py`
```

File: build-scripts/docker/repo-test/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    to see all supported releases
+
+## Elaborations
```
Comment (omoerbeek) [inline build-scripts/docker/repo-test/README.md:35]: A little note on how to use the generated files would be nice.

Comment (omoerbeek) [reply build-scripts/docker/repo-test/README.md:35]: e.g. use the `--test` argument. I now had to look into the PR to find that out.

**Review by omoerbeek — APPROVED** [review]
Instructions work here, two small nits in README

**Pull request merged**